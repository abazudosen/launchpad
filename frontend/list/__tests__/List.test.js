/* @flow */

import React from 'react';
import List from '../List';
import ReactShallowRenderer from 'react-test-renderer/shallow';

const renderer = new ReactShallowRenderer();

describe('List View', () => {
  test('renders empty', () => {
    const user = {
      id: 'test-ids',
      githubUsername: 'example',
      pads: [],
    };
    const component = renderer.render(<List user={user} onLogout={() => {}} />);
    expect(component).toMatchSnapshot();
  });

  test('renders full', () => {
    const user = {
      id: 'test-ids',
      githubUsername: 'example',
      pads: [
        {
          id: 'fakeid1',
          title: 'Fake 1',
          description: 'Description 1',
          code: 'code',
          deployedCode: 'deployed code',
          url: 'http://example.com',
          user: {
            id: 'test-ids',
            githubUsername: 'example',
            pads: [],
          },
          context: [
            {
              key: 'secret',
              value: 'secret',
            },
          ],
          dependencies: [
            {
              name: 'graphql-js',
              version: '0.8.9',
            },
          ],
          defaultQuery: '{ hello }',
          defaultVariables: null,
          token: 'test-token-2',
          draft: null,
        },
        {
          id: 'fakeid2',
          title: 'Fake 2',
          description: 'Description 2',
          code: 'code 2',
          deployedCode: 'deployed code 2',
          url: 'http://example.com/2',
          user: {
            id: 'test-ids',
            githubUsername: 'example',
            pads: [],
          },
          context: [
            {
              key: 'secret',
              value: 'secret',
            },
          ],
          dependencies: [
            {
              name: 'graphql-js',
              version: '0.8.9',
            },
          ],
          defaultQuery: '{ hello }',
          defaultVariables: null,
          token: 'test-token-2',
          draft: null,
        },
      ],
    };
    const component = renderer.render(
      <List
        user={{
          ...user,
          pad: [{}],
        }}
        onLogout={() => {}}
      />,
    );
    expect(component).toMatchSnapshot();
  });
});
