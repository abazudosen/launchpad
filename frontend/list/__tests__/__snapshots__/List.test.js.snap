// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty 1`] = `
<div
  className="List"
>
  <ListHeader
    onLogout={[Function]}
    user={
      Object {
        "githubUsername": "example",
        "id": "test-ids",
        "pads": Array [],
      }
    }
  />
  <div
    className="List-PadList"
  >
    
  </div>
</div>
`;

exports[`renders full 1`] = `
<div
  className="List"
>
  <ListHeader
    onLogout={[Function]}
    user={
      Object {
        "githubUsername": "example",
        "id": "test-ids",
        "pad": Array [
          Object {},
        ],
        "pads": Array [
          Object {
            "code": "code",
            "context": Array [
              Object {
                "key": "secret",
                "value": "secret",
              },
            ],
            "defaultQuery": "{ hello }",
            "defaultVariables": null,
            "dependencies": Array [
              Object {
                "name": "graphql-js",
                "version": "0.8.9",
              },
            ],
            "deployedCode": "deployed code",
            "description": "Description 1",
            "draft": null,
            "id": "fakeid1",
            "title": "Fake 1",
            "token": "test-token-2",
            "url": "http://example.com",
            "user": Object {
              "githubUsername": "example",
              "id": "test-ids",
              "pads": Array [],
            },
          },
          Object {
            "code": "code 2",
            "context": Array [
              Object {
                "key": "secret",
                "value": "secret",
              },
            ],
            "defaultQuery": "{ hello }",
            "defaultVariables": null,
            "dependencies": Array [
              Object {
                "name": "graphql-js",
                "version": "0.8.9",
              },
            ],
            "deployedCode": "deployed code 2",
            "description": "Description 2",
            "draft": null,
            "id": "fakeid2",
            "title": "Fake 2",
            "token": "test-token-2",
            "url": "http://example.com/2",
            "user": Object {
              "githubUsername": "example",
              "id": "test-ids",
              "pads": Array [],
            },
          },
        ],
      }
    }
  />
  <div
    className="List-PadList"
  >
    <ListPad
      pad={
        Object {
          "code": "code",
          "context": Array [
            Object {
              "key": "secret",
              "value": "secret",
            },
          ],
          "defaultQuery": "{ hello }",
          "defaultVariables": null,
          "dependencies": Array [
            Object {
              "name": "graphql-js",
              "version": "0.8.9",
            },
          ],
          "deployedCode": "deployed code",
          "description": "Description 1",
          "draft": null,
          "id": "fakeid1",
          "title": "Fake 1",
          "token": "test-token-2",
          "url": "http://example.com",
          "user": Object {
            "githubUsername": "example",
            "id": "test-ids",
            "pads": Array [],
          },
        }
      }
    />
    <ListPad
      pad={
        Object {
          "code": "code 2",
          "context": Array [
            Object {
              "key": "secret",
              "value": "secret",
            },
          ],
          "defaultQuery": "{ hello }",
          "defaultVariables": null,
          "dependencies": Array [
            Object {
              "name": "graphql-js",
              "version": "0.8.9",
            },
          ],
          "deployedCode": "deployed code 2",
          "description": "Description 2",
          "draft": null,
          "id": "fakeid2",
          "title": "Fake 2",
          "token": "test-token-2",
          "url": "http://example.com/2",
          "user": Object {
            "githubUsername": "example",
            "id": "test-ids",
            "pads": Array [],
          },
        }
      }
    />
  </div>
</div>
`;
